From 04232004a246387807dda2c38bdb743ac082f570 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Pawe=C5=82=20Lidwin?= <lidwinpawel@gmail.com>
Date: Sat, 15 Aug 2026 23:22:31 +0200
Subject: [PATCH 2/3] fix missing semicolon

---
 xuser.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dlls/xgameruntime/xuser.c b/dlls/xgameruntime/xuser.c
index 9b37413..108c481 100644
--- a/dlls/xgameruntime/xuser.c
+++ b/dlls/xgameruntime/xuser.c
@@ -381,7 +381,7 @@ static HRESULT load_endpoints( XUserHandle user, BYTE *buffer, SIZE_T size )
     IJsonObject *payload = NULL, *tmpObject = NULL;
     IJsonArray *jsonEndpoints = NULL, *jsonPolicies = NULL;
     IVector_IJsonValue *jsonEndpointsVector = NULL, *jsonPoliciesVector = NULL;
-    struct policy *policyCursor = NULL
+    struct policy *policyCursor = NULL;
     struct endpoint *endpointCursor = NULL;
     DOUBLE jsonNumber;
     UINT32 arraySize;
-- 
2.55.0

